packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -75,6 +77,20 @@ export class FileNavigatorWidget extends FileTreeWidget {
                 this.update();
             }
         });
+        this.enableMainPanelDnd();
+    }
+
+    protected enableMainPanelDnd(): void {
+        const mainPanelNode = this.shell.mainPanel.node;
+        this.addEventListener(mainPanelNode, 'drop', e => {
```

Comment:
I think listeners should be installed on the attach event and remove on detach otherwise they will be executed even if the navigator is not present.

Reply:
Do we unregister any of the listeners `onDetach` in the tree widgets? I cannot find it.

Reply:
I see. `this.addEventListener` pushes the listeners into the `toDisposeOnDetach` which will be disposed `onDetach`, hence I do not have to do anything with this.